test/jdk/java/lang/Math/Atan2Tests.java

```diff
@@ -23,44 +23,213 @@
+             * closest to pi/2.
+             */
+            {MIN_VALUE,  +0.0,        PI/2.0},
+            {MIN_NORM,   +0.0,        PI/2.0},
+            {1.0,        +0.0,        PI/2.0},
+            {MAX_VALUE,  +0.0,        PI/2.0},
+
+            {MIN_VALUE,  -0.0,        PI/2.0},
+            {MIN_VALUE,  -0.0,        PI/2.0},
+            {MIN_NORM,   -0.0,        PI/2.0},
+            {1.0,        -0.0,        PI/2.0},
+            {MAX_VALUE,  -0.0,        PI/2.0},
+
+            {InfinityD,  -MIN_VALUE,  PI/2.0},
+            {InfinityD,  -MIN_NORM,   PI/2.0},
+            {InfinityD,  -1.0,        PI/2.0},
+            {InfinityD,  -MAX_VALUE,  PI/2.0},
+
+            {InfinityD,  MIN_VALUE,   PI/2.0},
+            {InfinityD,  MIN_NORM,    PI/2.0},
+            {InfinityD,  1.0,         PI/2.0},
+            {InfinityD,  MAX_VALUE,   PI/2.0},
+
+            /*
+             * If the first argument is negative and the second argument is
+             * positive zero or negative zero, or the first argument is
+             * negative infinity and the second argument is finite, then the
+             * result is the double value closest to -pi/2.
+             */
+            {-MIN_VALUE,  +0.0,        -PI/2.0},
+            {-MIN_NORM,   +0.0,        -PI/2.0},
+            {-1.0,        +0.0,        -PI/2.0},
+            {-MAX_VALUE,  +0.0,        -PI/2.0},
+
+            {-MIN_VALUE,  -0.0,        -PI/2.0},
+            {-MIN_VALUE,  -0.0,        -PI/2.0},
+            {-MIN_NORM,   -0.0,        -PI/2.0},
+            {-1.0,        -0.0,        -PI/2.0},
+            {-MAX_VALUE,  -0.0,        -PI/2.0},
+
+            {-InfinityD,  -MIN_VALUE,  -PI/2.0},
+            {-InfinityD,  -MIN_NORM,   -PI/2.0},
+            {-InfinityD,  -1.0,        -PI/2.0},
+            {-InfinityD,  -MAX_VALUE,  -PI/2.0},
+
+            {-InfinityD,  MIN_VALUE,   -PI/2.0},
+            {-InfinityD,  MIN_NORM,    -PI/2.0},
+            {-InfinityD,  1.0,         -PI/2.0},
+            {-InfinityD,  MAX_VALUE,   -PI/2.0},
+
+            /*
+             * If both arguments are positive infinity, then the result is the
+             * double value closest to pi/4.
+             */
+            {InfinityD,  InfinityD,     PI/4.0},
+
+            /*
+             * If the first argument is positive infinity and the
+             * second argument is negative infinity, then the result
+             * is the double value closest to 3*pi/4.
+             */
+            {InfinityD,  -InfinityD,     3*PI/4.0},
+
+            /*
+             * If the first argument is negative infinity and the second
+             * argument is positive infinity, then the result is the double
+             * value closest to -pi/4.
+             */
+            {-InfinityD,  InfinityD,     -PI/4.0},
+
+            /*
+             * If both arguments are negative infinity, then the result is the
+             * double value closest to -3*pi/4.
```

Comment:
Perhaps add a note explaining that high precision computation shows that `-3*PI/4.0` is indeed the `double` closest to -3*pi/4.
Since `PI` is an approximation in the first place, and since there are other two operations (in particular, the multiplication by 3), the claim is not evident.

Reply:
Note (and manual double-check) added.